cmd/acr/purge.go

```diff
@@ -463,43 +468,41 @@ func getManifestsToDelete(ctx context.Context, acrClient api.AcrCLIClientInterfa
 		return nil, err
 	}
 	// This will act as a set if a key is present then it should not be deleted because it is referenced by a multiarch manifest
-	// that will not be deleted
-	doNotDelete := map[string]bool{}
-	candidatesToDelete := []acr.ManifestAttributesBase{}
-	// Iterate over all manifests to discover multiarchitecture manifests
+	// or the manifest has subjects attached
+	doNotDelete := set.New[string]()
+	var candidatesToDelete []acr.ManifestAttributesBase
 	for resultManifests != nil && resultManifests.ManifestsAttributes != nil {
 		manifests := *resultManifests.ManifestsAttributes
 		for _, manifest := range manifests {
-			if (*manifest.MediaType == manifestListContentType || *manifest.MediaType == manifestIndexContentType) && manifest.Tags != nil {
-				// If a manifest list is found and it has tags then all the dependent digests are
-				// marked to not be deleted.
-				var manifestListBytes []byte
-				manifestListBytes, err = acrClient.GetManifest(ctx, repoName, *manifest.Digest)
+			var cm customManifest
+			if _, ok := manifestMediaTypes[*manifest.MediaType]; ok {
+				var manifestBytes []byte
+				manifestBytes, err = acrClient.GetManifest(ctx, repoName, *manifest.Digest)
```

Comment:
```suggestion
				manifestBytes, err := acrClient.GetManifest(ctx, repoName, *manifest.Digest)
```

Reply:
done.

Reply:
> done.

It seems the code has not been changed yet.

Reply:
the suggested change will cause go-linter to throw shadow error from line 460. So I reused the `err` variable. Is there a best practice in this case?

Reply:
You may consider suppressing the lint for `err`. But I'm ok with the current state.